src/libs/antares/study/include/antares/study/parts/thermal/cluster.h

```diff
@@ -293,6 +293,16 @@ class ThermalCluster final: public Cluster, public std::enable_shared_from_this<
     */
     std::vector<double> PthetaInf;
 
+    bool hasMinGen() const
+    {
+        return anyMinGen_;
+    }
+
+    void anyMinGen(bool minGen)
+    {
+        anyMinGen_ = minGen;
+    }
+
```

Comment:
Use above `PthetaInf`, no need for a setter here

```cpp
class ThermalCluster
{
public:
    bool hasMinGen() const
    {
        if (!hasMinGen_.has_value())
        {
            hasMinGen_ = std::ranges::any_of(/*...*/);
        }
        return hasMinGen_.value();
    }

private:
    mutable std::optional<bool> hasMinGen_;
};

```

Reply:
Not used to see a getter with an **if** statement, it adds as bit of complexity.
Normally, `hasMinGen_` should be set at the cluster construction, so we would not need the if here.
I did the changes you ask though.